lightgbm/src/test/scala/com/microsoft/azure/synapse/ml/lightgbm/split1/VerifyLightGBMCommon.scala

```diff
@@ -74,4 +91,213 @@ class VerifyLightGBMCommon extends TestBase {
+      println(s"Training time, ${measurements(i).trainingTime}")
+      println(s"Overhead time, ${measurements(i).overheadTime}")
+      println(s"Task total times, ${measurements(i).taskTotalTimes.mkString(",")}")
+      println(s"Task overhead times, ${measurements(i).taskOverheadTimes().mkString(",")}")
+      println(s"Task initialization times, ${measurements(i).taskInitializationTimes().mkString(",")}")
+      println(s"Task library initialization times, ${measurements(i).taskLibraryInitializationTimes().mkString(",")}")
+      println(s"Task network initialization times, ${measurements(i).taskNetworkInitializationTimes().mkString(",")}")
+      println(s"Task data preparation times, ${measurements(i).taskDataPreparationTimes.mkString(",")}")
+      println(s"Task dataset wait times, ${measurements(i).taskWaitTimes().mkString(",")}")
+      println(s"Task dataset creation times, ${measurements(i).taskDatasetCreationTimes.mkString(",")}")
+      println(s"Task training iteration times, ${measurements(i).taskTrainingIterationTimes.mkString(",")}")
+      println(s"Task cleanup times, ${measurements(i).taskCleanupTimes().mkString(",")}")
+      println(s"** Completed Measurement $i")
+    })
+    println(s"***** Averaged results for $measurementCount runs")
+    var median = getMedian(measurements.map(m => m.totalTime))
+    println(s"Median Total time, $median")
+    median = getMedian(measurements.map(m => m.columnStatisticsTime))
+    println(s"Median Column statistics, $median")
+    median = getMedian(measurements.map(m => m.rowCountTime()))
+    println(s"Median Row count time, $median")
+    median = getMedian(measurements.map(m => m.samplingTime()))
+    println(s"Median Sampling time, $median")
+    median = getMedian(measurements.map(m => m.rowStatisticsTime()))
+    println(s"Median Row statistics time, $median")
+    median = getMedian(measurements.map(m => m.trainingTime))
+    println(s"Median Training time, $median")
+    median = getMedian(measurements.map(m => m.overheadTime))
+    println(s"Median Overhead time, $median")
+    var medianMax = getMedian(measurements.map(m => m.taskTotalTimes.max))
+    println(s"Median-max Task total times, $medianMax")
+    medianMax = getMedian(measurements.map(m => m.taskOverheadTimes().max))
+    println(s"Median-max Task overhead times, $medianMax")
+    medianMax = getMedian(measurements.map(m => m.taskInitializationTimes().max))
+    println(s"Median-max Task initialization times, $medianMax")
+    medianMax = getMedian(measurements.map(m => m.taskDataPreparationTimes.max))
+    println(s"Median-max Task data preparation times, $medianMax")
+    medianMax = getMedian(measurements.map(m => m.taskDatasetCreationTimes.max))
+    println(s"Median-max Task dataset creation times, $medianMax")
+    medianMax = getMedian(measurements.map(m => m.taskTrainingIterationTimes.max))
+    println(s"Median-max Task training iteration times, $medianMax")
```

Comment:
This looks like alot of duplicated code, and in previous PRs there is a ton of methods like this. Perhaps we should make these measurements just entries in a Map so that its more "dynamic" and we can iterate through it better